### PR TITLE
Check that we have mirror still to not panic and deadlock/hang

### PR DESCRIPTION
**Pull request by derekcollison**
Signed-off-by: Derek Collison <derek@nats.io>

/cc @nats-io/core

### CODE REVIEW

File: server/stream.go [hunk 1]
```diff
@@ -1139,6 +1139,10 @@ func (mset *stream) processMirrorMsgs() {
 
 	// Grab stream quit channel.
 	mset.mu.Lock()
+	if mset.mirror == nil {
```
Comment (kozlovic) [inline server/stream.go:1142]: and is it possible that mset.mirror.msgs be nil? because if so, then you would get a panic also because of that.

Comment (kozlovic) [reply server/stream.go:1142]: we don't apparently.

Comment (derekcollison) [reply server/stream.go:1142]: I double checked and we are good, I just nil out mset.mirror.

**Review by kozlovic — APPROVED** [review]
LGTM

**Pull request merged**